tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ProjectEncodingTest.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.tests.resources;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.eclipse.core.internal.resources.PreferenceInitializer;
+import org.eclipse.core.internal.resources.ValidateProjectEncoding;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.hamcrest.*;
+import org.junit.Test;
+
+/**
+ * Test for integration of marker
+ * {@link org.eclipse.core.resources.ResourcesPlugin#PREF_MISSING_ENCODING_MARKER_SEVERITY}.
+ */
+public class ProjectEncodingTest extends ResourceTest {
+
+	private static final int IGNORE = -1;
+
+	private IProject project;
+
+	@Override
+	protected void tearDown() throws Exception {
+		if (project != null) {
+			project.delete(true, null);
+		}
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).putInt(
+				ResourcesPlugin.PREF_MISSING_ENCODING_MARKER_SEVERITY,
+				PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT);
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).flush();
+		getWorkspace().getRoot().refreshLocal(IResource.DEPTH_INFINITE, null);
```

Comment:
What is that for?

Reply:
To wait until the workspace job finished. Without waiting the change-prefs tests will fail.

Reply:
> To wait until the workspace job finished. Without waiting the change-prefs tests will fail.

Can you try either:

```
TestUtil.waitForJobs(getName(), 10, 5_000, null);
```
Or:
```
TestUtil.waitForJobs(getName(), 10, 5_000, ValidateProjectEncoding.class);
```
(if that is the important job here)

Reply:
Thanks for the hint. Didn't know that class at all. I'll check.

Reply:
What does that test``TestUtil`` (contained in the test project for the resources plugin) better than the official API from ``JobManager``? I don't see any advantage right now.
* Using the official API is stable throughout 15 PR builds and all local builds w/ I ran before
* I need less parameters
* I don't need some local class that I can't use for testing any Eclipse plugin
* I can use an API that I can consider safely to be fully tested
** I can't find any unit tests for ``TestUtil``.

=> I very much would like to keep using the official job API here.

Reply:
> What does that test`TestUtil` (contained in the test project for the resources plugin) better than the official API from `JobManager`? I don't see any advantage right now.

I don't understand your question. Currently you are doing a refresh on the workspace (meaning, the test expects some I/O that Eclipse resources API didn't "see", and so a refresh will "sync" what the Eclipse API "sees" and the file system state). You said this is meant to wait for jobs.

Using the test utility to wait on jobs is the standard way of... waiting on jobs during tests. The utility exists since it will do some "nices to have" such as timeouts (necessary for tests that run into hanging jobs) as well as reporting which jobs are still running when the timeout occurs.

But its the same to me. If it starts failing due to whichever workspace job no longer needs a workspace lock, the test will be fixed.

Reply:
Sorry - I confused this with ``Job.getJobManager().join`` far below. Forget what I said about waiting. I'll check if I ever needed this line in the first place. My bad :)

Reply:
Yup. Dropped that line. I apparently had copied that line from ``NatureTest`` and didn't re-evaluate whether I need it. Thank you for your feedback on this!